src/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -107,60 +111,77 @@ private void clearImageCache() {
   /**
    * Return the appropriate unit image.
    */
-  public Image getImage(final UnitType type, final PlayerID player, final GameData data, final boolean damaged,
+  public Optional<Image> getImage(final UnitType type, final PlayerID player, final GameData data, final boolean damaged,
       final boolean disabled) {
     final String baseName = getBaseImageName(type, player, data, damaged, disabled);
     final String fullName = baseName + player.getName();
     if (m_images.containsKey(fullName)) {
-      return m_images.get(fullName);
+      return Optional.of(m_images.get(fullName));
     }
-    final Image baseImage = getBaseImage(baseName, player);
-    // We want to scale units according to the given scale factor.
-    // We use smooth scaling since the images are cached to allow
-    // to take our time in doing the scaling.
-    // Image observer is null, since the image should have been
-    // guaranteed to be loaded.
-    final int width = (int) (baseImage.getWidth(null) * m_scaleFactor);
-    final int height = (int) (baseImage.getHeight(null) * m_scaleFactor);
-    final Image scaledImage = baseImage.getScaledInstance(width, height, Image.SCALE_SMOOTH);
-    // Ensure the scaling is completed.
-    try {
-      Util.ensureImageLoaded(scaledImage);
-    } catch (final InterruptedException ex) {
-      ex.printStackTrace();
+    try (PerfTimer timer = Perf.startTimer("Load an image")) {
```

Comment:
The timer is good for spot checking stuff, but client side timing I'm rapidly coming to the opinion is without value. Will remove.